ui/src/main/webapp/app/gaffer/results-service.js

```diff
@@ -60,17 +60,38 @@ angular.module('app').factory('results', ['events', function(events) {
                 if(result.class === "uk.gov.gchq.gaffer.data.element.Entity") {
                     if(result.vertex !== undefined && result.vertex !== '') {
                         incomingResults.entities.push(result);
-                        results.entities.push(result);
+                        var toAdd = true;
+                        results.entities.forEach(function(element) {
+                            if(JSON.stringify(element) === JSON.stringify(result)) {
+                                toAdd = false;
+                            }
+                        });
+                        if (toAdd)
+                            results.entities.push(result);
                     }
                 } else if(result.class === "uk.gov.gchq.gaffer.data.element.Edge") {
                     if(result.source !== undefined && result.source !== ''
                     && result.destination !== undefined && result.destination !== '') {
                         incomingResults.edges.push(result);
-                        results.edges.push(result);
+                        var toAdd = true;
+                        results.edges.forEach(function(element) {
```

Comment:
Could this logic be pulled into a method to avoid repeating it?

Reply:
Latest push has moved it out into two linked functions.